CoreDistributedCache/NHibernate.Caches.CoreDistributedCache.Memcached/MemcachedFactory.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using Enyim.Caching;
+using Enyim.Caching.Configuration;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace NHibernate.Caches.CoreDistributedCache.Memcached
+{
+	/// <summary>
+	/// A Memcached distributed cache factory.
+	/// </summary>
+	public class MemcachedFactory : IDistributedCacheFactory
+	{
+		private static readonly INHibernateLogger Log = NHibernateLogger.For(typeof(MemcachedFactory));
+		private const string _configuration = "configuration";
+
+		private readonly IDistributedCache _cache;
+
+		/// <summary>
+		/// Constructor with configuration properties. It supports <c>configuration</c>, which has to be a JSON string
+		/// structured like the value part of the <c>"enyimMemcached"</c> property in an appsettings.json file.
+		/// </summary>
+		/// <param name="properties">The configurations properties.</param>
+		public MemcachedFactory(IDictionary<string, string> properties) : this()
+		{
+			MemcachedClientOptions options;
+			if (properties != null && properties.TryGetValue(_configuration, out var configuration) && !string.IsNullOrWhiteSpace(configuration))
+			{
+				options = JsonConvert.DeserializeObject<MemcachedClientOptions>(configuration);
+			}
+			else
+			{
+				Log.Warn("No {0} property provided", _configuration);
+				options = new MemcachedClientOptions();
+			}
+
+			var loggerFactory = new LoggerFactory();
+
+			_cache = new MemcachedClient(loggerFactory, new MemcachedClientConfiguration(loggerFactory, options));
```

Comment:
It is a bit messy to integrate a library meant to be used with the .Net Core default dependency injection system with NHibernate factories/configuration logic.

EnyimMemcachedCore was the worst case, due to its dependency on a logger factory (I have provided it through a wrapping of the NHibernate one) and its structured configuration (thus the JSON trick I have choosen for it).

Maybe should each factory also expose directly a constructor with the cache dependencies (at least `IMemcachedClientConfiguration` and maybe `ILoggerFactory` for this case)for users who would set the factory programmatically.